### DISCUSSION TITLE
Doesn't tools/misc/visualize_results.py shows the filtered predicted boxes?

### DISCUSSION THREAD

**Discussion by mauk95** [open-mmlab/mmdetection3d#2216] [Q&A]
I am using FCAF3D model for 3D object detection on Scannet dataset. I use the following command for evaluation:

python tools/misc/visualize_results.py configs/fcaf3d/fcaf3d_8x2_scannet-3d-18class.py --result /home/ahmed/Documents/mmdetection3d/data/scannet/results_all.pkl --show-dir /home/ahmed/Documents/mmdetection3d/data/scannet/show_results

However, the displayed open3d visual contains a lot of detection boxes as shown in the attached screenshot. The screenshot is for 'scene0568_00'. Also as seen in the other screenshot the gt contains 12 annotation boxes whereas, the predictions returned by the model contains 427 boxes.

Are these the predictions after applying NMS or not? Is there a script which shows the filtered predictions? If I just calculate the mAP for this scene it turns out 0.77 which is not understandable.

Also, the plotted ground truth and prediction values doesn't seem aligned with the point cloud 

![Screenshot from 2023-01-11 19-06-13](https://user-images.githubusercontent.com/17196636/211884059-3e72ecb0-b385-425a-b831-cf377ca7e16c.png)

![Screenshot from 2023-01-11 19-02-28](https://user-images.githubusercontent.com/17196636/211884111-37d16e0b-a1c9-4fa0-932a-2b17e6a382ab.png)

**Comment by JingweiZhang12** [comment]
It shows the final results, which are the predicted boxes after NMS. The number of bboxes is larger than the gt bboxes. It's normal since we should calculate the mAP with different Recall scores.  If you want to visualize them better,  you can filter the final results with a confidence score.
